daemons/pacemakerd/pacemakerd.c

```diff
@@ -365,6 +381,11 @@ main(int argc, char **argv)
     g_strfreev(processed_args);
     pcmk__free_arg_context(context);
 
-    pcmk__output_and_clear_error(error, NULL);
+    pcmk__output_and_clear_error(error, out);
+
+    if (out != NULL) {
+        out->finish(out, exit_code, true, NULL);
+        pcmk__output_free(out);
+    }
```

Comment:
For daemons, it might be a good idea to close the output as soon as we've processed any command-line calls -- in this case, just after freeing old_instance would be a good spot. That would help ensure we don't try to output anything past that point.

Reply:
Done.